workflow/controller/operator.go

```diff
@@ -1460,16 +1461,21 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 			node = lastChildNode
 		} else {
 			// Create a new child node and append it to the retry node.
-			nodeName = fmt.Sprintf("%s(%d)", retryNodeName, len(retryParentNode.Children))
+			childrenLength := strconv.Itoa(len(retryParentNode.Children))
```

Comment:
We don't need  this line at all. In the following line, `Sprintf` will do the conversion for us and in line 1475 we can do as suggested